### PR TITLE
[147] Ensure path sanity checks provide clear errors during registrat…

### PR DESCRIPTION
**Pull request by peterver**
…ion checks

## Checklist

- [v] I have ensured my pull request is not behind the main or master branch of the original repository.
- [v] I have rebased all commits where necessary so that reviewing this pull request can be done without having to merge it first.
- [v] I have written a commit message that passes commitlint linting.
- [v] I have ensured that my code changes pass linting tests.
- [v] I have ensured that my code changes pass unit tests.
- [v] I have described my pull request and the reasons for code changes along with context if necessary.

### CODE REVIEW

**Comment by peterver** [comment]
this is an adjusted version of #147 with unit tests added for correctness

**Comment by etroynov** [comment]
@peterver hi you can change [v] -> [x] and list will be look like here:
https://github.com/koajs/router/pull/153

**Comment by etroynov** [comment]
@wengeezhang could you check it?

File: lib/router.js [hunk 1]
```diff
@@ -197,9 +197,14 @@ for (let i = 0; i < methods.length; i++) {
         name = null;
       }
 
-      this.register(path, [method], middleware, {
-        name: name
-      });
+      // Sanity check to ensure we have a viable path candidate (eg: string|regex|non-empty array)
+      if (
+        typeof path !== "string" &&
```
Comment (etroynov) [inline lib/router.js:202]: nit: In project usually use single quotes not double

File: test/lib/router.js [hunk 2]
```diff
@@ -993,7 +993,30 @@ describe('Router', function () {
           done();
         });
     });
+    
+    it ('correctly returns an error when not passed a path for verb-specific registration (gh-148)', function () {
+      const app = new Koa();
```
Comment (etroynov) [inline test/lib/router.js:998]: Hi,

In this test you have some unused variables, you can fix it like this:
```
const router = new Router();

 try {
  router.get(function () {});
 } catch (e) {
   expect(e.message).to.be('You have to provide a path when adding a get handler');
 }
 ```

**Comment by peterver** [comment]
> @peterver I suppose it's not good approach make PR from `master` to `master`, maybe you should create separate branch for it.

Apologies, first contribution here and want to get going :D I'm used to working in a git-flow approach (develop/master) and wasn't sure about the proper practice here :). Will adhere for future PR's thanks!

**Comment by peterver** [comment]
@etroynov I've opened a new #155 with all of the above adjustment requests included :)

**Pull request closed without merging**